### DISCUSSION TITLE
No event name for Page View

### DISCUSSION THREAD

**Discussion by ImpNpkRC** [duracelltomi/gtm4wp#363] [Ideas]
Is it possible to add event_name to the main page load Data Layer push? Or that feature is already there and I was not able to enable it?
Having "message" as the event name is not helpful when comes to setting tags that rely on page load parameters to send it to GA4.

![image](https://github.com/user-attachments/assets/a8f5ee7c-d93e-4af5-8a5c-d1a762429f7d)

**Comment by duracelltomi** [comment]
There is not event name because those variables are normally available once the contanier has been loaded. In your screenshot, the Container Loaded event happens earlier, may I ask if you added the GTM container code manually to your site?

**Reply by ImpNpkRC** [reply]
> There is not event name because those variables are normally available once the contanier has been loaded. In your screenshot, the Container Loaded event happens earlier, may I ask if you added the GTM container code manually to your site?

Hey, Thomas.

Thank you for the reply. GTM in this case is being deployed through the Plugin under General > Google Tag Manager ID, so I'm not sure why the sequence is not being followed.

It makes sense if those variables would load before the container is loaded then I'd be able to collect them and pass them into other tags,

**Reply by duracelltomi** [reply]
Can you share a website URL?

**Reply by ImpNpkRC** [reply]
> Can you share a website URL?

Sure. It's an internal site for testing purposes so you may see other unrelated and not following best practices there.

https://leviathan.napkyn.com/
